UTIKit/UTI.swift

```diff
@@ -247,8 +247,28 @@ public struct UTI: CustomStringConvertible, CustomDebugStringConvertible, Equata
 
 }
 
-public func ==(lhs: UTI, rhs: UTI) -> Bool {
-    return UTTypeEqual(lhs.utiString as CFString, rhs.utiString as CFString)
+extension UTI: Equatable {
+	static public func ==(lhs: UTI, rhs: UTI) -> Bool {
+		// Since there is no UTTypeHash, or anything like that, we're forced to implement hashable
+		// according to raw hashing of `utiString`. To ensure our implementation of `Hashable` and
+		// `Equatable` are logically valid, we need to assert that `==` and `UTTypeEqual` behave
+		// equally
+		let expected = UTTypeEqual(lhs.utiString as CFString, rhs.utiString as CFString)
+		let actual = lhs.utiString == rhs.utiString
```

Comment:
A quick check shows that `UTTypeEqual` is not a simple string match. It seems to be case-insensitive.

```swift
UTTypeEqual("public.image" as CFString, "public.image" as CFString) // true
UTTypeEqual("public.image" as CFString, "PUBLIC.IMAGE" as CFString) // true
```

However, I couldn't find any documentation on how it was implemented.

Reply:
Getting a canonical identifier from the declaration may work. You also need to fall back to `utiString` for dynamic UTI.

```swift
extension UTI: Hashable {
    public func hash(into hasher: inout Hasher) {
        hasher.combine(self.declaration.identifier ?? self.utiString.lowercased())
    }
}
```

Reply:
Hey, thanks for getting back on this! Really busy right now, but I'll fix these up when I get a chance and repsuh!